lib/jarvis/thread_logger.rb

```diff
@@ -0,0 +1,42 @@
+require "cabin"
+
+module Jarvis module ThreadLogger
+  class State
+    def initialize
+      @state = {}
+      @lock = Mutex.new
+    end
+
+    def <<(hash)
+      @lock.synchronize do
+        @state[Thread.current] = hash
```

Comment:
This is called inside a thread, since we are running this on a thread pool, `Thread.current` value could be reused between calls?

 Wouldn't it better to generate an ID for a Command Execution to reference the log and use bounded data structure?